pallets/parachain-staking/src/tests.rs

```diff
@@ -9061,7 +9121,7 @@ fn test_on_initialize_weights() {
 			//
 			// following this assertion, we add individual weights together to show that we can
 			// derive this number independently.
-			let expected_on_init = 3_017_871_000;
+			let expected_on_init = 2_506_497_000;
```

Comment:
@notlesh This is the second time I've had to touch this test and honestly the only thing I do is make the number match the expectation. Could we somehow improve the test so it doesn't require us to update the magic number all the time? I'm afraid that we'd just ignore the failure and always update numbers without realizing what actually went down. 

In my case, admittedly, I'd started adding the weight from auto-compounding to `extra_weight`, and here changed the execution behavior. Even though I knew in both cases what changed I couldn't come up with a good magic number value. :(

Reply:
I agree that it's annoying. The point of it is so that we don't accidentally change the overall weight of on_init without realizing it. If you can think of a better way to accomplish that, I'm all for it.

Otherwise I think we could discuss whether it is worth having it at all. I like it for the reason above, but not enough that it should annoy everyone :)

While we're on the subject, I'm surprised that the amount decreased so much...